homeassistant/components/vacuum/__init__.py

```diff
@@ -147,6 +147,11 @@ def battery_level(self):
         """Return the battery level of the vacuum cleaner."""
         return None
 
+    @property
+    def battery_icon(self):
+        """Return the battery icon for the vacuum cleaner."""
+        raise NotImplementedError()
```

Comment:
Can't we leave here a default icon ?

Reply:
This is implemented in both `VacuumEntity` and `VacuumStateEntity`.